app/viewers/embed/viewer/file.rb

```diff
@@ -36,6 +36,17 @@ def display_file_search?
                                  @purl_object.contents.map(&:files).flatten.length >= min_files_to_search
       end
 
+      def message
+        return { type: 'embargo', message: embargo_message } if @purl_object.embargoed?
+
+        if @purl_object.location_restriction
+          return { type: 'location-restricted',
+                   message: I18n.t('restrictions.restricted_access', location: @purl_object.restricted_location) }
```

Comment:
Might it help to have a fallback empty string in the case where there is a code that (for some reason) does not map to what we have in settings.  That will probably not the case, but may be good to have something just in case.

Reply:
Yeah I was trying to figure out what the best way of doing this is. Because I would think we would want the error or something to display wrong if the cocina is wrong. @andrewjbtw Do you have a preference?

Reply:
This is in case a new location is introduced in cocina and used before this code is updated? Cocina validation should prevent an item from being published with a code not on the list.

If we can assume that an item with an unexpected code (for banner purposes) is still valid Cocina, I would fall back on a generic message rather than an error since the new rights setting should be active and supported in Stacks by the time someone uses it. 

> Access is restricted to on-site visitors to the Stanford Libraries. See Access conditions for more information.